src/riscv/lib/src/state_backend/proof_backend/merkle.rs

```diff
@@ -28,89 +28,111 @@ pub const MERKLE_LEAF_SIZE: NonZeroUsize = NonZeroUsize::new(4096).unwrap();
 /// [`DynArrays`]: [`crate::state_backend::layout::DynArray`]
 pub const MERKLE_ARITY: usize = 4;
 
-/// Turns a [MerkleTree] into a [CompressedMerkleTree]
-fn merkle_tree_to_compressed_merkle_tree(merkle_tree: MerkleTree) -> CompressedMerkleTree {
-    use CompressedMerkleTree::Leaf as CompressedLeaf;
-    use CompressedMerkleTree::Node as CompressedNode;
-
-    impl_modify_map_collect(
-        merkle_tree,
-        |subtree| match subtree {
-            MerkleTree::Leaf(hash, access_info, data) => {
-                ModifyResult::LeafStop((hash, access_info, data))
+fn maybe_compress_node(hash: Hash, children: Vec<CompressedMerkleTree>) -> CompressedMerkleTree {
```

Comment:
This feels more appropriate as an associated function of `CompressedMerkleTree`.

```
impl CompressedMerkleTree {
    fn node(hash: Hash, children: Vec<Self>) -> Self {
        todo!()
    }
}